corehq/apps/locations/bulk_management.py

```diff
@@ -878,6 +863,20 @@ def iter_unprocessed_ancestor_ids(stubs):
                     loc_object.parent = loc.new_parent
                     loc_object.save()
 
+    _seen = set()
+
+    def iter_unprocessed_ancestor_ids(stubs):
+        # Returns a generator of all ancestor IDs of locations in 'stubs' which
+        # haven't already been returned in a previous call
+        for loc in stubs:
+            if not loc.is_new:
+                pk = loc.db_object.pk
+                while pk is not None and pk not in _seen:
+                    _seen.add(pk)
+                    location = old_locations_by_pk[pk]
+                    yield location.location_id
+                    pk = location.parent_id
+
```

Comment:
Made a comment on the commit, but it didn't show up in diff view, so I'll copy here:

This change is twofold:

It's difficult to assign a parent location at the time when the location stub is being created. You'd need to either reorder locations to be top-to-bottom first (which is hard, since they're not yet fully parsed), or do two passes, one to init everything, then a second to add parents. I didn't like the second option since you'd have to mutate the location after its construction. Luckily, it looked like very little actually requires the parent object - pretty much just at the point of saving, so I opted to move the lookup there and drop the linkage.

The second aspect of this is that I realized we should be looking for the old ancestors here anyways, since they're the ones that are affected by a deletion.

Reply:
Thanks Ethan... Can you add couple of comments within `save_locations` to understand whats happening here better.